src/lib/zxing/index.ts

```diff
@@ -0,0 +1,27 @@
+import { BrowserQRCodeReader, IScannerControls } from '@zxing/browser'
+import { DecodeContinuouslyCallback } from '@zxing/browser/esm/common/DecodeContinuouslyCallback'
+
+export class Scanner {
+  codeReader!: BrowserQRCodeReader
+  videoElement: HTMLVideoElement
+
+  constructor({ videoElement }: { videoElement: HTMLVideoElement }) {
+    this.videoElement = videoElement
+  }
+
+  async init() {
+    const { BrowserCodeReader, BrowserQRCodeReader } = await import('@zxing/browser')
+
+    this.codeReader = new BrowserQRCodeReader()
+
+    return new (BrowserCodeReader.listVideoInputDevices as any)()
```

Comment:
Remove this line. Add `getCameras` method to this class.

```ts
async getCameras() {
  const { BrowserCodeReader } = await import('@zxing/browser')
  return BrowserCodeReader.listVideoInputDevices()
}
```

Reuse that method in `QrcodeScannerDialog.vue`:

```ts
const init = async () => {
// ...
  await scanner.value.init()
  cameras.value = await scanner.getCameras()
```